src/puppetlabs/puppetdb/amq_migration.clj

```diff
@@ -136,31 +137,36 @@
           (finally
             (.acknowledge message)))))))
 
-(defn consume-everything
-  "Attempts to read all available messages from ActiveMQ, giving up
-  after 5 seconds if one is not available"
-  [^MessageConsumer consumer process-message id]
-  (try
-    (loop []
-      (when-let [msg (.receive consumer 5000)]
-        (process-message (swap! id inc) msg)
-        (recur)))
-    (catch javax.jms.IllegalStateException e
-      (log/info (i18n/trs "Received IllegalStateException, shutting down")))
-    (catch Exception e
-      (log/error e))))
-
 (defn transfer-mq-messages
+  "Returns true if it's likely that all of the existing messages were
+  transferred."
   [^ConnectionFactory conn-pool endpoint process-message id]
   (with-open [connection (create-connection conn-pool)
               session (create-session connection)
+              producer (.createProducer session (create-queue session endpoint))
               consumer (.createConsumer session (create-queue session endpoint))]
     (log/info (i18n/trs "Processing messages from the Queue: {0}" endpoint))
-    (consume-everything consumer process-message id)))
+    (mq/send-message! connection endpoint
+                      "\"ignore me\""
+                      {"command" "end-of-line"})
+    (try
+      (loop []
+        (when-let [msg (.receive consumer 120000)]
```

Comment:
Would it be useful here to log something after a period of time? This will cause startup to hang for 2 minutes. I can't think of a reason WHY it would ever take that long, but if it did. There's no indication here what's going on. Any opinions here @wkalt ?